docs/code_conventions/README.MD

```diff
@@ -1,131 +1,349 @@
 ```
 
-## Arguments
-When naming an argument, the name should be all lowercase with spaces replaced by underscores. See the example below:
+## Main function 
+- Create `main` function and in the main extract all the integration parameters
+- Implement **_command** function for each integration command (e.g. `say_hello_command(client, demisto.args())`)
+- Wrap the commands with try/except in the main, in order to handle the exceptions properly. `return_error()` receives error message and returns error entry back into Demisto. It will also print the full error to the Demisto logs. 
+- For logs use `demisto.info()`, `demisto.debug()`, and `LOG()` functions.
+- In the main create the Client instance and pass that client to other functions
+```python
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    username = demisto.params().get('credentials').get('identifier')
+    password = demisto.params().get('credentials').get('password')
 
-### Do this:
-```
-!cylance-get-incident incident_id
-```
- ### Not this:
- ```python
-!cylance-get-incident incidentId
-```
+    # Remove trailing slash to prevent wrong URL path to service
+    server_url = demisto.params()['url'][:-1] \
+        if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
 
-Additionally, arguments are to be passed through the ```sample_function_command``` function and must not reside in the primary ```sample_function``` function. Observe the following:
+    verify_certificate = not demisto.params().get('insecure', False)
 
-**Do This**
+    # How many time before the first fetch to retrieve incidents
+    first_fetch_time = demisto.params().get('fetch_time', '3 days')
 
-```python
-def sample_function(arg1):
-    response = arg1 + 'is the argument'
-    return response
-
-def sample_function_command():
-    arg1 = demisto.args().get('arg1')
-    response = sample_function(arg1)
-    demisto.results('This is the response - {}'.format(response))
+    proxies = handle_proxy()
+
+    LOG('Command being called is %s' % (demisto.command()))
+
+    try:
+        client = Client(server_url, verify_certificate, username, password, proxies)
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            return_outputs(result, None)
+
+        elif demisto.command() == 'fetch-incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            next_run, incidents = fetch_incidents(
+                client=client,
+                last_run=demisto.getLastRun(),
+                first_fetch_time=first_fetch_time)
+
+            demisto.setLastRun(next_run)
+            demisto.incidents(incidents)
+
+        elif demisto.command() == 'helloworld-say-hello':
+            return_outputs(*say_hello_command(client, demisto.args()))
+
```

Comment:
shouldnt we have general return_outputs here?

Reply:
I am not using our template here, but simple helloworld example